### PR TITLE
Toggle clippy::pedantic on neqo-crypto.

### PR DESCRIPTION
**Pull request by o0Ignition0o**
Part of #150, I was struggling trying to disable the default features (which doesn't do too well with build.rs) but given the number of warnings, we might as well clear them all instead :)

I don't know enough about the internals to be 100% confident Aead::expansion() will remain static, so please let me know if I should revert the change and allow unused self instead :)

### CODE REVIEW

**Comment by o0Ignition0o** [comment]
cannot find function `max_value` in module `std::u64`
It seems target related, I'm going to find out which one, and fix it :)

File: neqo-crypto/src/aead.rs [hunk 1]
```diff
@@ -59,7 +59,8 @@ impl Aead {
         unsafe { Self::from_raw(version, cipher, s, prefix) }
     }
 
-    pub fn expansion(&self) -> usize {
+    #[must_use]
+    pub fn expansion() -> usize {
```
Comment (martinthomson) [inline neqo-crypto/src/aead.rs]: Removing `&self` here might be something that clippy wants, but I'm not sure that I do.

File: neqo-crypto/src/time.rs [hunk 2]
```diff
@@ -200,6 +200,10 @@ mod test {
     }
 
     #[test]
+    // We allow replace_consts here beceuse
```
Comment (martinthomson) [inline neqo-crypto/src/time.rs]:
```suggestion
    // We allow replace_consts here because
```

File: neqo-crypto/src/aead.rs (hunk 1 above)
Comment (o0Ignition0o) [reply neqo-crypto/src/aead.rs]: Yeah I'm not really sure about that either 😁

**Pull request merged**